Add glossary page (#29)
vshn/application-catalog-docs: docs/modules/ROOT/pages/reference/glossary.adoc

= Glossary

== General

=== Service

> A managed application that can be ordered, is auomatically provisioned and can be consumed by a service consumer.

Reviewer: ```suggestion
An application that can be ordered.
It is automatically provisioned, can be consumed by a service consumer and generally adheres to our quality requirements.
```

=== Service Consumer

> A customer-owned application that connects to a service as a client.

Reviewer: ```suggestion
An application that consumes a service instance.
```

I wouldn't focus on "customer", it can be anyone, even another system component.

Author: Agree, but I wouldn't use `consume` again, since that is the term we want to define ;)

=== Service Order
The act of deploying a Kubernetes resource that triggers the provisioning of a new service instance.

=== Service Instance

> An instance of a service offering.

Reviewer: We haven't defined "service offering". I'd say that it should be like that:

```suggestion
An instance of a service.
```

> A single isolated ocurrurance of a service.

Reviewer 2: ```suggestion
A single isolated occurrence of a service.
```

==== Cloud Instance

> An instance of cloud service offering by a cloud provider.

Reviewer: ```suggestion
An instance of a cloud service offering by a cloud service provider.
```

==== Converged Service Instance
An instance of a converged service blueprint.
Either in the same namespace as the service order was made or in another namespace.

== Service Location

Services can be provisioned on various places:

=== Converged Cluster

> Service runs on the same cluster as the service consumer.

Reviewer: ```suggestion
Service instance runs on the same cluster as the service consumer.
```

=== Service Cluster
Service runs on one or more dedicated clusters, reserved for services.

> Usually deployed via a Helm Chart.

Reviewer: ```suggestion
```

Maybe skip that Helm specific here.

=== Cloud Service

> A type of service architecture offered by a cloud provider such as AWS or GCP.

Reviewer: ```suggestion
A type of service architecture offered by a cloud service provider such as AWS or GCP.
```

=== Converged Service
A type of service developed to run on a converged cluster.

> Usually deployed via a Helm Chart.

Reviewer: ```suggestion
```

> ==== Converged Service Architectures

Reviewer: Maybe a sentence which says what's meant with "architecture"?

Standalone::
A single node instance which is neither replicated nor clustered.

Replicated::
The service offers data replication to one or more other instances.

Clustered::
A fully clustered version of the service, usually offering active / active replication with fully automated cluster handling.

== Personas

=== Service User
A human that interacts with {product}.

=== Service Engineer
A human that engineers and maintains a service (code).

> === Support Engineer

Reviewer: ```suggestion
=== Service Operator
```

Author: Isn't that ambiguous with Operator from Kubernetes? I imagine that could lead to confusions.
What if we define "Service Engineer" as the one doing Ops and the "Service Maintainer" as the "programmer behind the scenes"?

Reviewer: Oh yes, that could be misleading, right. I like the suggestion :+1:
